irodsadmin/management/commands/irodsorphans.py

```diff
@@ -25,184 +26,226 @@
+            for p in Project.objects.all().order_by('full_title')
+        ]
+
+    def _is_zone(self, collection):
+        """
+        Check if a given collection matches the format of path to a landing zone
+        collection.
+        """
+        return '/landing_zones/' in collection.path and re.search(
+            r'^\d{8}_\d{6}', collection.name
+        )
+
+    def _is_assay_or_study(self, collection):
+        """
+        Check if a given collection matches the format of path to a study or assay
+        collection.
+        """
+        return re.match(
+            r'(assay|study)_[a-f0-9]{8}-([a-f0-9]{4}-){3}[a-f0-9]{12}',
+            collection.name,
         )
-    ]
-
-
-def get_project_collections(irods_backend):
-    """Return a list of all study collection names."""
-    return [irods_backend.get_path(p) for p in Project.objects.all()]
-
-
-def is_zone(collection):
-    """
-    Check if a given collection matches the format of path to a landing zone
-    collection.
-    """
-    return '/landing_zones/' in collection.path and re.search(
-        r'^\d{8}_\d{6}', collection.name
-    )
-
-
-def is_assay_or_study(collection):
-    """
-    Check if a given collection matches the format of path to a study or assay
-    collection.
-    """
-    return re.match(
-        r'(assay|study)_[a-f0-9]{8}-([a-f0-9]{4}-){3}[a-f0-9]{12}',
-        collection.name,
-    )
-
-
-def is_project(collection):
-    """
-    Check if a given collection matches the format of path to a project
-    collection.
-    """
-    return re.search(
-        r'projects/([a-f0-9]{2})/\1[a-f0-9]{6}-([a-f0-9]{4}-){3}[a-f0-9]{12}$',
-        collection.path,
-    )
-
-
-def get_orphans(irods, irods_backend, expected, assays):
-    """
-    Return a list of orphans in a given irods session that are not in a given
-    list of expected collections.
-    """
-    orphans = []
-    collections = irods.collections.get('/{}/projects'.format(irods.zone))
-
-    for collection in irods_backend.get_colls_recursively(collections):
-        if (
-            is_zone(collection)
-            or is_assay_or_study(collection)
-            or is_project(collection)
-        ):
-            if collection.path not in expected:
-                orphans.append(collection.path)
-
-    for assay in assays:
-        if not assay.get_plugin():
-            continue
-        with irods_backend.get_session() as irods:
-            for collection in irods_backend.get_child_colls(
-                irods, irods_backend.get_path(assay)
-            ):
-                if collection.path not in expected:
-                    orphans.append(collection.path)
-    return orphans
 
+    def _is_project(self, collection):
+        """
+        Check if a given collection matches the format of path to a project
+        collection.
+        """
+        return re.search(
+            r'projects/([a-f0-9]{2})/\1[a-f0-9]{6}-([a-f0-9]{4}-){3}[a-f0-9]{12}$',
```

Comment:
Similar to my other comment, we should be ensuring the entire start of the path instead of just looking "projects" or the UUID somewhere within the path, as users can technically create arbitrarily named collections within the path.

The regexes should search for things under the projects path retrieved with `IrodsAPI.get_projects_path()`, adding the begining of the line character `^` to make sure we are evaluating the correct part of the path. Maybe this was already wrong in the original implementation, in that case I've simply missed it before.